### PR TITLE
feat(client): Go client for identities

### PR DESCRIPTION
**Pull request by benhoyt**
This adds the Go client types and methods for identities (part of [spec OP043](https://docs.google.com/document/d/1nASgUt-piV94i1cpFsbEPRk_xFbZt8AskgL97gh1Dgg/edit)), specifically:

* `Identity` and `IdentityAccess` types, with access level constants
* `Identities`
* `AddIdentities`
* `UpdateIdentities`
* `ReplaceIdentities`
* `RemoveIdentities`

### CODE REVIEW

File: client/identities.go [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+		Path:   "/v1/identities",
+	})
+	if err != nil {
+		return nil, err
+	}
+	var identities map[string]*Identity
+	err = resp.DecodeResult(&identities)
+	if err != nil {
+		return nil, err
+	}
```
Comment (hpidcock) [inline client/identities.go:72]: We should probably be kind to ourselves and check the server did not return a nil entry in the map.

File: client/notices_test.go [hunk 2]
```diff
@@ -42,6 +42,8 @@ func (cs *clientSuite) TestNotice(c *C) {
 	}}`
 	notice, err := cs.cli.Notice("123")
 	c.Assert(err, IsNil)
+	c.Assert(cs.req.Method, Equals, "GET")
```
Comment (hpidcock) [inline client/notices_test.go:45]: Is this just adding consistency to our tests?

**Review by hpidcock — APPROVED** [review]
LGTM with a bit of defensive programming with the return value from the server.

File: client/identities.go (hunk 1 above)
Comment (benhoyt) [reply client/identities.go:72]: Fair enough -- added this and a test.

File: client/notices_test.go (hunk 2 above)
Comment (benhoyt) [reply client/notices_test.go:45]: Yep, that's right -- and specifically that the client is sending the right HTTP method.

**Pull request merged**